File: 프로그래머스/0/181909. 접미사 배열/접미사 배열.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(string my_string) {
    vector<string> s;
    for(int i = 0; i < my_string.length(); i++) {
        s.push_back(my_string.substr(i));
    }
    
    sort(s.begin(), s.end());
    
    return s;
}

File: 프로그래머스/0/181909. 접미사 배열/접미사 배열.java
import java.util.*;

class Solution {
    public String[] solution(String my_string) {
        ArrayList<String> answer = new ArrayList<>();
        
        for (int i = 0; i < my_string.length(); i++) {
            answer.add(my_string.substring(i));
        }
        
        Collections.sort(answer);
        
        String[] result = new String[answer.size()];
        for (int i = 0; i < answer.size(); i++)
            result[i] = answer.get(i);
        
        return result;
    }
}